Add chapter to migration guide on how to migrate to Volto

pbauer opened the pull request.

pbauer committed:
Add chapter to migration guide on how to migrate to Volto

stevepiercy committed:
Rename file with dashes, fix grammar and syntax, add a few inline todos.

stevepiercy committed:
Add to new file to index

stevepiercy committed:
Add a todo for the backend section to include an introduction
Add some html_meta tags

stevepiercy committed:
Use MyST syntax for link.

stevepiercy reviewed (changes requested):
Because I renamed this file to use dashes instead of underscores, and there were too many itty-bitty changes, I just pushed changes to this PR.

However I have several questions and comments that we need to resolve before I feel comfortable merging this PR. I wrote them inline as `todo`.

pbauer committed:
update migrate-to-volto

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
+
+Technically it is a migration since the version of Plone does not get changed and it is no upgrade-step.
+It could be called a upgrade though because Volto is the new default frontend.
+Not 100% sure...
```
I think that "upgrade" is less correct than "migration". In the instructions, we "Migrate RichText fields to Volto blocks" and the form's URI contains "migrate_to_volto". Let's use migrate instead of upgrade everywhere in this document, including its file name.

Admittedly it is a little bit weird because it is located within the "upgrading" directory, but I'm OK with a URI of `/backend/upgrading/migrate-to-volto.html`.

pbauer replied:
Ok, I'm fine with using the term Migration for this :)

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
-I think there needs to be an introduction to this chapter to give it better context. Here are a few questions I have when I jump to this document from who knows where.
+Plone 6 comes with a new default frontend, called Volto.
+Volto is written in React and uses `plone.restapi` to communicate with the backend.
+When creating a new site in Plone 6 you have the option to choose between creating a Volto site and a Classic Plone site.
```
Out of curiosity, what is the actual terminology used in the UI?

In my mind, one creates a Plone site with a frontend of either Volto or Classic UI. I dislike the phrase of "creating a Volto site" or "creating a Classic Plone site" as it implies that they are two totally different sites instead of just UIs. Both options create a "Plone site". Only the UI is the primary difference.

```suggestion
When creating a new Plone 6 site, you may choose between Volto and Classic UI for the frontend.
```

pbauer replied:
I actually is "Create a new Plone site" and "Create Classic Plone site":

<img width="571" alt="Bildschirmfoto 2022-05-17 um 19 06 53" src="https://user-images.githubusercontent.com/453208/168870925-5f93b9af-87c4-424a-868a-b2f871f9aa77.png">

I'll update accordingly

stevepiercy replied:
Where would be a good place to discuss this UI? I think it could be improved. I approach this with fresh eyes, and it is... confusing.

For this PR, aligning with the UI is the way forward.

stevepiercy replied:
Found it, and not easily.

https://github.com/plone/Products.CMFPlone/commit/cd57b6e82fca76a35caac018a18211e7a7dddc39

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
+Plone 6 comes with a new default frontend, called Volto.
+Volto is written in React and uses `plone.restapi` to communicate with the backend.
+When creating a new site in Plone 6 you have the option to choose between creating a Volto site and a Classic Plone site.
+This is because there are some non-trivial differences between the configuration of a Classic Plone and a Volto site.
```
```suggestion
This choice is presented because there are some non-trivial differences between their configurations.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
-1.  What is the target audience?
-1.  What are the pre-requisites?
-1.  Other?
+This document discusses these differences and tells admins and developers how to modify their existing Plone site to be compatible with Volto.
```
```suggestion
This document discusses these differences.
It also informs administrators and developers of how to migrate their existing Plone 6 site with Classic UI for its frontend to instead become compatible with Volto for its frontend.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
-1.  What are the pre-requisites?
-1.  Other?
+This document discusses these differences and tells admins and developers how to modify their existing Plone site to be compatible with Volto.
+Pre-requisites: The site needs to be upgraded to Plone 6 before
```
````suggestion
```important
As a pre-requisite, your Plone site must be [upgraded to Plone 6](v60) before migrating to Volto for the frontend.
```
````

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
````diff
@@ -10,54 +10,54 @@ html_meta:
 
-If you provide bullet points, I can write up the narrative and push to this PR.
-```
+Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing site to make it compatible with Volto.
````
```suggestion
Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing Plone 6 site to make it compatible with Volto.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
````diff
@@ -10,54 +10,54 @@ html_meta:
-If you provide bullet points, I can write up the narrative and push to this PR.
-```
+Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing site to make it compatible with Volto.
+You can access this form in the browser when you are logged-in as an administrator by opening `http://localhost:8080/Plone/@@migrate_to_volto` (where `localhost` is your server, `8080` the port that Plone is running on and `Plone` is the name of the Plone instance).
````
```suggestion
You can access this form in the browser when you are logged in as an administrator by opening `http://localhost:8080/Plone/@@migrate_to_volto` (where `localhost` is your server, `8080` is the port on which Plone is running, and `Plone` is the name of the Plone instance).
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
+Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing site to make it compatible with Volto.
+You can access this form in the browser when you are logged-in as an administrator by opening `http://localhost:8080/Plone/@@migrate_to_volto` (where `localhost` is your server, `8080` the port that Plone is running on and `Plone` is the name of the Plone instance).
+
+After upgrading an existing site to Plone 6 (see {doc}`v60`) there will also appear a message **You can prepare your site for Volto, the default frontend of Plone 6!** with a link to that form.
```
```suggestion
Additionally, after upgrading an existing site to Plone 6 (see {doc}`v60`) a message will appear, **You can prepare your site for Volto, the default frontend of Plone 6!**, with a link to that form.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
 
-A migrated site will still work with Plone Classic UI, but it will behave differently.
-For example, editors can only effectively work with the content using Volto.
+A site that is make compatible with Volto will be accessible with Plone Classic UI, but it will behave differently.
```
```suggestion
A site that is made compatible with Volto will be accessible with Plone Classic UI, but it will behave differently.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
-A migrated site will still work with Plone Classic UI, but it will behave differently.
-For example, editors can only effectively work with the content using Volto.
+A site that is make compatible with Volto will be accessible with Plone Classic UI, but it will behave differently.
+For example, editors can only effectively work with the content using Volto because html is no longer editable in the TinyMCE editor.
```
```suggestion
For example, editors can only effectively work with the content using Volto because HTML is no longer editable in the TinyMCE editor used only in Classic UI.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -10,54 +10,54 @@ html_meta:
 
 We could either use inline literals as shown in the suggestion, or we could link to their Glossary entry (or create one), GitHub repo, PyPI project, or relevant chapter in the docs. Which do you think is best?
+
+I like `{py:mod}` because it is semantic and renders like a inline-literal. We don't require link in these cases.
```
Thanks for the explanation, and that makes sense to me now. Please delete the todo discussion for this one.

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
````diff
@@ -102,3 +102,10 @@ The required steps are:
 It is recommended to use the default settings, but you can choose to skip some of the migration steps in the form.
+
+```{note}
+If you are migrating a existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site
````
```suggestion
If you are migrating an existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site, then you do not need to use the form `@@migrate_to_volto`.
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
````diff
@@ -102,3 +102,10 @@ The required steps are:
+
+```{note}
+If you are migrating a existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site
+you do not need to use the form `@@migrate_to_volto`.
````
Merge with previous line.

```suggestion
```

stevepiercy commented on docs/backend/upgrading/upgrade-to-volto.md:
```diff
@@ -102,3 +102,10 @@ The required steps are:
+If you are migrating a existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site
+you do not need to use the form `@@migrate_to_volto`.
+
+All the changes documented above can be done efficiently during export and import and are discussed [here](https://github.com/collective/collective.exportimport/issues/133).
```
```suggestion
All the changes documented above can be done efficiently during export and import.
[Read details](https://github.com/collective/collective.exportimport/issues/133).
```

stevepiercy reviewed (changes requested):
Mostly English grammar and syntax fixes, with a couple of questions and comments. I think this document should use "migrate" instead of "upgrade" throughout.

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
Update docs/backend/upgrading/upgrade-to-volto.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

pbauer committed:
update docs on migrating to volto

pbauer committed:
update naming according to add-site form

pbauer committed:
small fixes

mauritsvanrees commented on docs/backend/upgrading/migrate-to-volto.md:
```diff
@@ -0,0 +1,97 @@
+    In Volto adding Collections is disabled by default.
+    Instead folderish pages with listing blocks are used.
+    This step turns all collections into folderish pages.
+    The criteria of the collection are uses to configure a listing block in that page.
```
uses -> used

stevepiercy replied:
Fixed in c1d559abe18885f39411b9b9dbb6577512f31111, along with other things to reduce some review cycles.

mauritsvanrees reviewed (dismissed):
Very clear, thanks. 
One typo.

stevepiercy committed:
Add Glossary terms for Volto and Classic UI.
Use shorter sentences.
More English grammar, syntax, and spelling cleanup.
Fix some MyST syntax.
Add guilabel for GUI elements.

stevepiercy committed:
Merge branch '6-dev' into upgrade_to_volto

stevepiercy reviewed (approved):
I made several changes in c1d559abe18885f39411b9b9dbb6577512f31111. They should be reviewed by another pair of eyes.

pbauer commented:
LGTM
